test/performance/io/charconv_benchmark.cpp

```diff
@@ -0,0 +1,257 @@
+// ==========================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ==========================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert, FU Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <benchmark/benchmark.h>
+
+#include <cstring>
+#include <sstream>
+#include <string_view>
+
+#include <seqan3/std/charconv>
+
+// -----------------------------------------------------------------------------
+// to_char
+// -----------------------------------------------------------------------------
+template <typename arithmetic_type>
+static void to_char(benchmark::State& state)
+{
+    arithmetic_type val{120};
+    std::stringstream ss;
+    char buffer[10];
+
+    for (auto _ : state)
+    {
+        std::memset(buffer, 0, sizeof(buffer));
+        std::to_chars(&buffer[0], &buffer[0] + sizeof(buffer), val);
+        ss << buffer;
```

Comment:
You might want to clear ss.rdbuf regularly so that it doesn't dynamically grow a buffer internally. I suspect it would be faster and show the influence of the actual operation more clearly, but I am not 100% sure.
Also, do you have to clear buffer everytime? It will be overwritten anyways, or not?

Reply:
> Also, do you have to clear buffer every time? It will be overwritten anyway, or not?

In this case, I would not have to clear the buffer as I always read the same number. But if this should be implemented to generically read to chars, I will not know if I read a smaller (in terms of digits) value afterwards and I do need to clear the buffer every time. So clearing it here I think better resembles the use case.

Reply:
> You might want to clear ss.rdbuf regularly so that it doesn't dynamically grow a buffer internally. I suspect it would be faster and show the influence of the actual operation more clearly, but I am not 100% sure.

Clearing the buffer decreases runtime.